cmd/juju/application/deploy.go

```diff
@@ -1267,6 +1286,16 @@ func (c *DeployCommand) maybeReadCharmstoreBundleFn(apiRoot DeployAPI) func() (d
 	}
 }
 
+func (c *DeployCommand) getCharmStoreAPIURL(conAPIRoot api.Connection) (string, error) {
+	_, backend := base.NewClientFacade(conAPIRoot, "Controller")
```

Comment:
This doesn't look right.
To get a controller facade client, we just do:
controller.NewClient(root)